spec/queries/check_for_content_item_preventing_draft_from_being_published_spec.rb

```diff
@@ -0,0 +1,138 @@
+      it "returns nil" do
+        expect(subject).to eq(nil)
+      end
+    end
+
+    context "with a single content item" do
+      before do
+        FactoryGirl.create(:content_item,
+          content_id: content_id,
+          base_path: base_path,
+          user_facing_version: 1,
+          locale: "en",
+        )
+      end
+
+      include_examples "check succeeds"
+    end
+
+    context "with two content items of different locales" do
+      before do
+        FactoryGirl.create(:content_item,
+          content_id: content_id,
+          base_path: base_path + ".en",
+          user_facing_version: 1,
+          locale: "en",
+        )
+
+        FactoryGirl.create(:content_item,
+          content_id: content_id,
+          base_path: base_path + ".es",
+          user_facing_version: 1,
+          locale: "es",
+        )
+      end
+
+      include_examples "check succeeds"
+    end
+
+    context "with a unpublished item, of type \"substitute\", and a draft at the same base path" do
+      before do
+        FactoryGirl.create(:unpublished_content_item,
+          content_id: SecureRandom.uuid,
+          base_path: base_path,
+          unpublishing_type: "substitute",
+          user_facing_version: 1,
+          locale: "en",
+        )
+
+        FactoryGirl.create(:content_item,
+          content_id: content_id,
+          base_path: base_path,
+          user_facing_version: 1,
+          locale: "en",
+        )
+      end
+
+      include_examples "check succeeds"
+    end
+
+    context "with a published item, with a substitutable document_type, and a draft at the same base path" do
+      before do
+        FactoryGirl.create(:content_item,
+          content_id: SecureRandom.uuid,
+          base_path: base_path,
+          state: "published",
+          document_type: "unpublishing",
+          user_facing_version: 1,
+          locale: "en",
+        )
+
+        FactoryGirl.create(:content_item,
+          content_id: content_id,
+          base_path: base_path,
+          user_facing_version: 1,
+          locale: "en",
+        )
+      end
+
+      include_examples "check succeeds"
+    end
+
+    context "with a unpublished item, and a draft at the same base path" do
+      before do
+        @blocking_content_item = FactoryGirl.create(:gone_unpublished_content_item,
+          content_id: SecureRandom.uuid,
+          base_path: base_path,
+          user_facing_version: 1,
+          locale: "en",
+        )
+
+        FactoryGirl.create(:content_item,
+          content_id: content_id,
+          base_path: base_path,
+          user_facing_version: 1,
+          locale: "en",
+        )
+      end
+
+      it "fails, returning the id of the content item" do
+        expect(subject).to eq(@blocking_content_item.id)
+      end
+    end
+
+    context "with a published item, and a draft at the same base path" do
+      before do
+        @blocking_content_item = FactoryGirl.create(:content_item,
```

Comment:
`before` makes sense when you are performing an action for it's side effects, whereas let(!) is the predominant approach used across the test suite for setting variables that are used in tests.